### PR TITLE
Add mime types for sharedlib and executable

### PR DESCRIPTION
**Pull request by sharidas**
Add  mime types for application/x-sharedlib
and application/x-executable

Signed-off-by: Sujith H <sharidasan@owncloud.com>

## Description

Add mime type mapping for `application/x-sharedlib` and `application/x-executable`. This would help files detected with the mentioned missing mime types. Hence this pr adds `application/x-sharedlib` and `application/x-executable` mime types to `mimetypemapping.dist.json` file.

## Related Issue

- Fixes <issue_link>

## Motivation and Context

Add missing mime types mapping to the mimetypemapping.dist.json file. The missing mime types detected are `application/x-sharedlib` and `application/x-executable`.

## How Has This Been Tested?

- Enable firewall app.
- Navigate to  Admin -> Settings -> security page. Add a group with name `sharedlib block` 
- Select `File mimetype upload` from the drop down list and from the select drop down select `is`. Add `application/x-sharedlib` to the text field.
- Select `Blocked Requests` only` from the drop down next to `Save Rules` button. Click the button `Save Rules`.
- Now upload a file `libau.so`.
- In the log file user should see :
```
{"reqId":"pSacFi2oCtqQNupse4gN","level":2,"time":"2018-11-20T12:12:32+00:00","remoteAddr":"::1","user":"admin","app":"firewall","method":"PUT","url":"\/testing3\/remote.php\/dav\/files\/admin\/libau.so","message":"Firewall blocked request due to a matching rule \"sharedlib block\""}
```

## Screenshots (if appropriate):
- `.so` file 
sharedlib
- `.dll` file 
dll

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

**Comment by sharidas** [comment]
- [ ] Should we address files without extension? ( context : https://github.com/owncloud/enterprise/issues/2955#issuecomment-440199384 )
  - [ ] The files without extension, like executables, which can have `application/x-sharedlib` mime type.
  - [ ] The files without extension may also have `application/x-executable` mime type. 
Eg: ```printf '\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\2\0' > /tmp/file```
The file `/tmp/file` will have mime type `application/x-executable`.

File: resources/config/mimetypemapping.dist.json [hunk 1]
```diff
@@ -179,5 +179,6 @@
 	"xrf": ["image/x-dcraw"],
 	"yaml": ["application/yaml", "text/plain"],
 	"yml": ["application/yaml", "text/plain"],
-	"zip": ["application/zip"]
+	"zip": ["application/zip"],
+	"so": ["application/x-sharedlib"]
```
Comment (PVince81) [inline resources/config/mimetypemapping.dist.json]: please also add "dll"

**Comment by sharidas** [comment]
As per the discussion at https://github.com/owncloud/core/pull/33595#discussion_r241717344, I have added
- [x] dll support.

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by PVince81** [comment]
@sharidas please backport

**Comment by sharidas** [comment]
Backport to stable10 PR https://github.com/owncloud/core/pull/33893

**Pull request merged**